src/PresentationalComponents/Labels/CategoryLabel.js

```diff
@@ -19,43 +20,29 @@ const CategoryLabel = ({ labelList, isCompact }) => {
     labelList.map((label) => label.id)
   );
 
-  const label = (icon, text) => (
-    <Label icon={icon} variant="outline" color="blue" isCompact={isCompact}>
-      {text}
-    </Label>
-  );
-  const labels = (id, isCompact) => {
-    if (id === 1) {
-      return label(
-        <AutomationIcon />,
-        intl.formatMessage(messages.availability)
-      );
-    }
-    if (id === 2) {
-      return label(
-        <LockIcon />,
-        intl.formatMessage(messages.security),
-        isCompact
-      );
-    }
-    if (id === 3) {
-      return label(
-        <CubeIcon />,
-        intl.formatMessage(messages.stability),
-        isCompact
-      );
-    }
-    if (id === 4) {
-      return label(
-        <PortIcon />,
-        intl.formatMessage(messages.performance),
-        isCompact
-      );
-    }
-  };
+  const labels = (id) =>
+    [
+      <LabelIcon
+        icon={<AutomationIcon />}
+        text={intl.formatMessage(messages.availability)}
+      />,
+      <LabelIcon
+        icon={<LockIcon />}
+        text={intl.formatMessage(messages.security)}
+      />,
+      <LabelIcon
+        icon={<CubeIcon />}
+        text={intl.formatMessage(messages.stability)}
+      />,
+      <LabelIcon
+        icon={<PortIcon />}
+        text={intl.formatMessage(messages.performance)}
+      />,
+    ][id];
+
   return (
-    <LabelGroup numLabels={1} isCompact={isCompact}>
-      {sortedFrequencyList.map((id) => labels(Number(id[0]), isCompact))}
+    <LabelGroup numLabels={1} isCompact>
+      {sortedFrequencyList.map((id) => labels(Number(id[0] - 1)))}
```

Comment:
Using a number here to identify a specific label is still a bit odd. Maybe there is a way that we can rewrite that at some point to have the `labels` in an object.

Reply:
I admit its a little weird,  but it works well. I do think it'll be worth a visit down the line, especially if it gets used more.